### PR TITLE
Lab/4

### PR DESCRIPTION
**Pull request by artegful**
https://artegful.github.io/itirod/

### CODE REVIEW

File: js/Page.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            .then((response) => response.text())
+            .then((text) => 
+            {
+                this._html = text;
+            });
+    }
+
+    async Show(callback)
+    {
+        if (this._html == null)
```
Comment (nastik1811) [inline js/Page.js]: use `===`

File: js/Page.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    async Load()
+    {
+        await fetch(`./pages/${this._name}.html`)
+            .then((response) => response.text())
+            .then((text) => 
+            {
+                this._html = text;
+            });
+    }
+
```
Comment (nastik1811) [inline js/Page.js]: If you use the async/await syntax, you can do the same more concisely
```
  const response = await fetch(`./pages/${this._name}.html`);
  this._html = await response.text();
```

File: js/Page.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        this._container.innerHTML = this._html;
+        this.AfterShowingHook();
+
+        callback();
+    }
+
+    AfterShowingHook()
+    { }
+
+    async Hide(callback)
```
Comment (nastik1811) [inline js/Page.js]: Why do you need async keyword here?

**Review by nastik1811 — CHANGES_REQUESTED** [review]
I don't understand the difference between different modes (they look the same for me). Maybe, you can explain
image
image

**Comment by artegful** [comment]
> I don't understand the difference between different modes (they look the same for me). Maybe, you can explain image image

The first picture shows the logo and the button, that leads to the page with typing. Haven't found a good logo icon, so I've decided just to reuse the keyboard icon, but bigger.
And about the modes:
time mode - counter to the top-right of the text shows the remaining time in seconds, when it hits zero - game ends.
words mode - you have to type all the words, counter shows the amount remaining.

File: js/Page.js (hunk 1 above)
Comment (artegful) [reply js/Page.js]: got it

File: js/Page.js (hunk 2 above)
Comment (artegful) [reply js/Page.js]: got it

File: js/Page.js (hunk 3 above)
Comment (artegful) [reply js/Page.js]: Yeah, not needed, changed the method body, forgot to remove async

**Pull request merged**